.github/workflows/ci-build.yaml

```diff
@@ -377,12 +379,15 @@ jobs:
     timeout-minutes: 15 # must be strictly greater than the timeout in .golangci.yml
     env:
       GOPATH: /home/runner/go
+      USE_NIX: true
```

Comment:
mmm this changes if/how/when things get installed, I don't think it should be on during CI.

In the `Makefile` you can just check if Nix exists before running `./hack/patch-nix-hash.sh`. Like check if the `nix` command exists, e.g. `command -v nix`. Much simpler check 😅